### PR TITLE
Update XLM-RoBERTa model documentation with enhanced usage examples and improved layout

### PR DESCRIPTION
**Pull request by AshAnand34**
# What does this PR do?

This PR creates a model card for the model XLM-RoBERTa as per the guidelines in #36979.

* Reorganized shield icons for better visual alignment.
* Added detailed usage examples for masked language modeling using both `Pipeline` and `AutoModel`.
* Updated model overview and key features for clarity and completeness.
* Included links to original checkpoints and relevant resources for user convenience.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/xlm-roberta.md [hunk 1]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
-">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+<div style="float: right;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md:17]: Badges should go before `# XLM-RoBERTa`

File: docs/source/en/model_doc/xlm-roberta.md [hunk 2]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
+        <img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+    </div>
 </div>
 
-## Overview
+[XLM-RoBERTa](https://arxiv.org/abs/1911.02116) is a large multilingual language model trained on 2.5TB of filtered CommonCrawl data across 100 languages. What makes XLM-RoBERTa unique is its ability to achieve strong performance on both high and low-resource languages without sacrificing per-language performance, outperforming previous multilingual models like mBERT by significant margins.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]: Link to the huggingface.co/papers instead of arxiv

```suggestion
[XLM-RoBERTa](https://huggingface.co/papers/1911.02116) is a large multilingual masked language model trained on 2.5TB of filtered CommonCrawl data across 100 languages. It shows that scaling the model provides strong performance gains on high-resource and low-resource languages. The model uses the [RoBERTa](./roberta) pretraining objectives on the [XLM](./xlm) model.
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 3]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
 
-The XLM-RoBERTa model was proposed in [Unsupervised Cross-lingual Representation Learning at Scale](https://arxiv.org/abs/1911.02116) by Alexis Conneau, Kartikay Khandelwal, Naman Goyal, Vishrav Chaudhary, Guillaume
-Wenzek, Francisco Guzmán, Edouard Grave, Myle Ott, Luke Zettlemoyer and Veselin Stoyanov. It is based on Facebook's
-RoBERTa model released in 2019. It is a large multi-lingual language model, trained on 2.5TB of filtered CommonCrawl
-data.
+You can find all the original XLM-RoBERTa checkpoints under the [XLM-RoBERTa](https://huggingface.co/models?search=xlm-roberta) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
You can find all the original XLM-RoBERTa checkpoints under the [Facebook AI community](https://huggingface.co/FacebookAI) organization.
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 4]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
-languages, using more than two terabytes of filtered CommonCrawl data. Our model, dubbed XLM-R, significantly
-outperforms multilingual BERT (mBERT) on a variety of cross-lingual benchmarks, including +13.8% average accuracy on
-XNLI, +12.3% average F1 score on MLQA, and +2.1% average F1 score on NER. XLM-R performs particularly well on
-low-resource languages, improving 11.8% in XNLI accuracy for Swahili and 9.2% for Urdu over the previous XLM model. We
-also present a detailed empirical evaluation of the key factors that are required to achieve these gains, including the
-trade-offs between (1) positive transfer and capacity dilution and (2) the performance of high and low resource
-languages at scale. Finally, we show, for the first time, the possibility of multilingual modeling without sacrificing
-per-language performance; XLM-R is very competitive with strong monolingual models on the GLUE and XNLI benchmarks. We
-will make XLM-R code, data, and models publicly available.*
+The example below demonstrates how to use XLM-RoBERTa for masked language modeling with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 5]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
 
-This model was contributed by [stefan-it](https://huggingface.co/stefan-it). The original code can be found [here](https://github.com/pytorch/fairseq/tree/master/examples/xlmr).
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-## Usage tips
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md:40]: See comment [here](https://github.com/huggingface/transformers/pull/38597#discussion_r2130231240)

File: docs/source/en/model_doc/xlm-roberta.md [hunk 6]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
+# Example in French
+result = unmasker("Bonjour, je suis un modèle [MASK].")
+print(result)
+```
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```python
+from transformers import AutoModelForMaskedLM, AutoTokenizer
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md:57]: See comment [here](https://github.com/huggingface/transformers/pull/38597#discussion_r2130234157)

File: docs/source/en/model_doc/xlm-roberta.md [hunk 7]
```diff
@@ -16,43 +16,93 @@ rendered properly in your Markdown viewer.
...
+    outputs = model(**inputs)
+    predictions = outputs.logits.argmax(dim=-1)
+
+# Decode prediction
+predicted_token = tokenizer.decode(predictions[0][inputs["input_ids"][0] == tokenizer.mask_token_id])
+print(f"Predicted token: {predicted_token}")
+```
+
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]: See your other PR for the same comments (such as quantization example and command line example) to apply here!

**Review by stevhliu — COMMENTED** [review]
Good start, but check your other PR for some of the same comments you should apply here!

File: docs/source/en/model_doc/xlm-roberta.md (hunk 7 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md]: Just added a quantization example and CLI example.

File: docs/source/en/model_doc/xlm-roberta.md (hunk 6 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md:57]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md (hunk 5 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md:40]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md (hunk 4 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md (hunk 3 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md (hunk 2 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md (hunk 1 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md:17]: Fixed

File: docs/source/en/model_doc/xlm-roberta.md [hunk 8]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
-  language from the input ids.
-- Uses RoBERTa tricks on the XLM approach, but does not use the translation language modeling objective. It only uses masked language modeling on sentences coming from one language.
+[XLM-RoBERTa](https://huggingface.co/papers/1911.02116) is a large multilingual masked language model trained on 2.5TB of filtered CommonCrawl data across 100 languages. It shows that scaling the model provides strong performance gains on high-resource and low-resource languages. The model uses the [RoBERTa](./roberta) pretraining objectives on the [XLM](./xlm) model.
+
+You can find all the original XLM-RoBERTa checkpoints under the [Facebook AI community](https://huggingface.co/FacebookAI) organization.
+
+> [!TIP]
+> Click on the XLM-RoBERTa models in the right sidebar for more examples of how to apply XLM-RoBERTa to different cross-lingual tasks like classification, translation, and question answering.
+
+The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`], [`AutoModel`], and from the command line.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
The example below demonstrates how to predict the `<mask>` token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 9]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```python
+import torch
+from transformers import pipeline
+
+# Initialize the pipeline with XLM-RoBERTa
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 10]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+pipeline = pipeline(
+    task="fill-mask",
+    model="facebook/xlm-roberta-base",
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
    model="FacebookAI/xlm-roberta-base",
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 11]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+    torch_dtype=torch.float16,
+    device=0
+)
+# Example in French
+pipeline("Bonjour, je suis un modèle [MASK].")
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
pipeline("Bonjour, je suis un modèle <mask>.")
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 12]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+```
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```python
+from transformers import AutoModelForMaskedLM, AutoTokenizer
+import torch
+
+# Load model and tokenizer
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 13]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+tokenizer = AutoTokenizer.from_pretrained(
+    "facebook/xlm-roberta-base"
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
    "FacebookAI/xlm-roberta-base"
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 14]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+)
+model = AutoModelForMaskedLM.from_pretrained(
+    "facebook/xlm-roberta-base",
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
    "FacebookAI/xlm-roberta-base",
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 15]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+    torch_dtype=torch.float16,
+    device_map="auto",
+    attn_implementation="sdpa"
+)
+
+# Prepare input
+inputs = tokenizer("Bonjour, je suis un modèle [MASK].", return_tensors="pt").to("cuda")
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
inputs = tokenizer("Bonjour, je suis un modèle <mask>.", return_tensors="pt").to("cuda")
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 16]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+
+masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
+predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
+predicted_token = tokenizer.decode(predicted_token_id)
+
+print(f"The predicted token is: {predicted_token}")
+```
+
+</hfoption>
+<hfoption id="CLI">
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
<hfoption id="transformers CLI">
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 17]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+
+```bash
+transformers-cli run fill-mask \
+    --model facebook/xlm-roberta-base \
+    --text "Bonjour, je suis un modèle [MASK]." \
+    --device 0
+# Output: The predicted token is: modèle
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
echo -e "Plants create <mask> through a process known as photosynthesis." | transformers-cli run --task fill-mask --model FacebookAI/xlm-roberta-base --device 0
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 18]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+```
+</hfoption>
+</hfoptions>
+
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [quantization guide](../quantization) overview for more available quantization backends.
+
+The example below uses bitsandbytes the quantive the weights to 4 bits
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
The example below uses [bitsandbytes](../quantization/bitsandbytes) to quantize the weights to 4 bits.
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 19]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+
+```python
+import torch
+from transformers import AutoModelForMaskedLM, AutoTokenizer, BitsAndBytesConfig
+# Load model and tokenizer with bitsandbytes quantization
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 20]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+tokenizer = AutoTokenizer.from_pretrained("facebook/xlm-roberta-base")
+model = AutoModelForMaskedLM.from_pretrained(
+    "facebook/xlm-roberta-base",
+    torch_dtype=torch.float16,
+    device_map="auto",
+    attn_implementation="flash_attention_2",
+    quantization_config=quantization_config
+)
+
+inputs = tokenizer("Bonjour, je suis un modèle [MASK].", return_tensors="pt").to("cuda")
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
inputs = tokenizer("Bonjour, je suis un modèle <mask>.", return_tensors="pt").to("cuda")
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 21]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+from transformers import AutoModelForMaskedLM, AutoTokenizer, BitsAndBytesConfig
+# Load model and tokenizer with bitsandbytes quantization
+
+quantization_config = BitsAndBytesConfig(
+    load_in_4bit=True,
+    bnb_4bit_compute_dtype=torch.bfloat16
+    bnb_4bit_quant_type="nf4",  # or "fp4" for float 4-bit quantization
+    bnb_4bit_use_double_quant=True,  # use double quantization for better performance
+)
+tokenizer = AutoTokenizer.from_pretrained("facebook/xlm-roberta-base")
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
tokenizer = AutoTokenizer.from_pretrained("facebook/xlm-roberta-large")
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 22]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+model = AutoModelForMaskedLM.from_pretrained(
+    "facebook/xlm-roberta-base",
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
    "facebook/xlm-roberta-large",
```

File: docs/source/en/model_doc/xlm-roberta.md [hunk 23]
```diff
@@ -14,45 +14,124 @@ rendered properly in your Markdown viewer.
...
+outputs = model.generate(**inputs, max_new_tokens=100)
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+```
+
+## Notes
+
+- Unlike some XLM models, XLM-RoBERTa doesn't require `lang` tensors to understand what language is being used. It automatically determines the language from the input IDs
+- Uses RoBERTa's training approach but adapted for multilingual data
+- Particularly effective for low-resource languages
+- Supports all standard RoBERTa tasks (classification, token classification, QA, etc.)
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.md]:
```suggestion
- Unlike some XLM models, XLM-RoBERTa doesn't require `lang` tensors to understand what language is being used. It automatically determines the language from the input IDs.
```

**Review by stevhliu — COMMENTED** [review]
Some additional comments about using the correct mask token and checkpoint

File: docs/source/en/model_doc/xlm-roberta.md (hunk 16 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm-roberta.md]: fixed

**Review by stevhliu — APPROVED** [review]
Thanks!
